include/picongpu/fields/incidentField/profiles/DispersiveLaser.def

```diff
@@ -0,0 +1,107 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Anton Helm, Rene Widera,
+ *                     Richard Pausch, Alexander Debus, Sergei Bastrakov,
+ *                     Fabia Dietrich, Klaus Steiniger
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/fields/incidentField/profiles/BaseParam.def"
+
+namespace picongpu
+{
+    namespace fields
+    {
+        namespace incidentField
+        {
+            namespace profiles
+            {
+                namespace defaults
+                {
+                    struct DispersiveLaserParam : public BaseParam
+                    {
+                        /** Beam waist: distance from the axis where the pulse intensity (E^2)
+                         *              decreases to its 1/e^2-th part,
+                         *              at the focus position of the laser
+                         * W0_SI = FWHM_of_Intensity / sqrt{ 2* ln(2) }
+                         *                             [   1.17741    ]
+                         *
+                         * unit: meter
+                         */
+                        static constexpr float_64 W0_SI = 5.0e-6 / 1.17741;
+
+                        /** The laser pulse will be initialized PULSE_INIT times of the PULSE_LENGTH
+                         * WATCH OUT! Dispersion parameters may lead to pulse lenghtening
+                         * Please ensure to choose a value high enough to cover the whole pulse,
```

Comment:
Would it make sense to provide a small python script that tells you the minimal required `PULSE_INIT` for a given set of dispersion parameters and a given time step etc, so that the users don't need to figure it out themselves?

Reply:
Additionally, I think it would be useful if we have separate parameters for the size of the DFT and the initialization delay. If someone wants to use these lasers for e.g. optical probing at later timesteps in the simulation, it is a bad idea to be forced to increase the size of the DFT from the dispersive laser implementation... But this might be another issue/pull request, since it also affects other incident field laser implementations. The only way to implement a time-delay with the `GaussianBeam` laser is the respective `PULSE_INIT`, which is not really the most convenient variable for this usage.

Reply:
Absolutely agree, we should introduce a variable 'PULSE_DELAY' or so in all laser fields, to be able to delay the laser pulse arrival in the simulation box with respect to simulation start.
But this should be part of a separate PR.

If you agree, I will open an issue and assign to myself.

Reply:
This would be great 👍🏻

Reply:
It would be great to have a separate PR for the `PULSE_DELAY` feature. But I think it is still a problem, that `PULSE_INIT` is used for 2 different things in this implementation and it would be better to have a separate variable for the size of the DFT.

Reply:
Independent of the introduction of `PULSE_DELAY` in a separate PR, `PULSE_INIT` and the number of points in the DFT are inseparable. `PULSE_INIT` defines the size of a time window within which we need to know the field values at specified instances of time given by the time step `DELTA_T`. The size of the window and time step together fix the required total number of samples in time domain, which by definition of the Fourier transform is exactly the number of samples required in frequency domain.